packages/syft/src/syft/core/adp/scalar/intermediate_gamma_scalar.py

```diff
@@ -127,7 +127,7 @@ def max_lipschitz_via_explicit_search(
 
         r2_diffs = np.array(
             [
-                GammaScalar(x.min_val, x.value, x.max_val, entity=x.entity).poly  # type: ignore
+                GammaScalar(x.min_val, x.value, x.max_val, entity=x.entity, prime=x.prime).poly  # type: ignore
```

Comment:
Q: I think the input scalars are type 
```Type[PhiScalars, GammaScalars]```
https://github.com/OpenMined/PySyft/blob/c3c913e225cb9a2fbd2c04a9727efb8e367b2395/packages/syft/src/syft/core/adp/scalar/abstract/intermediate_scalar.py#L59
I think the PhiScalars do not have .prime attribute,which would raise an error, 
or the input scalar are only ```GammaScalars``` in the scenario.

Reply:
@rasswanth-s I think we are assuming the input scalars to be of the `GammaScalar` type, cause we're are recreating the GammaScalar from the input scalars (and prime is a required parameter to do so). Maybe we add a check and filter out `GammaScalars` before calculating `r2_diffs`.

Reply:
Ahh ok , should we throw an Error if there is  `PhiScalar` (Or we do get a PhiScalar at this point) ? I am not  very comfortable with DP part of the codebase.